Penteract/Assets/Scripts/HUDController.cpp

```diff
@@ -340,11 +362,88 @@ void HUDController::UpdateFangCooldowns(GameObject* fangSkillCanvas) {
 				}
 			}
 		}
+
 		++skill;
 	}
 }
 
-void HUDController::UpdateOnimaruCooldowns(GameObject* onimaruSkillCanvas) {
+
+
+void HUDController::PlayCoolDownEffect(AbilityRefreshEffect* ef, Cooldowns cooldown) {
+	if (ef != nullptr) {
+		ef->Play();
+		abilityCoolDownsRetreived[cooldown] = true;
+	}
+}
+
+
+void HUDController::AbilityCoolDownEffectCheck(Cooldowns cooldown, GameObject* canvas) {
+	if (!abilityCoolDownsRetreived[cooldown]) {
+		if (canvas) {
+			if (cooldowns[cooldown] == 1.0f) {
+
+				AbilityRefreshEffect* ef = nullptr;
+				switch (cooldown) {
```

Comment:
You don't need this switch case. Every case does the same except for the index of the GetChildren. Might be better to have the size as index and then do a generic if to check if it has a children. Same for Onimaru.

Reply:
More generic version implemented.